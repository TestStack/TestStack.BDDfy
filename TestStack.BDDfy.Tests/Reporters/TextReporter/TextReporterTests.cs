using System;
using System.Runtime.CompilerServices;
using System.Text;
using ApprovalTests;
using NUnit.Framework;
using TestStack.BDDfy.Reporters;

namespace TestStack.BDDfy.Tests.Reporters.MarkDown
{
    [TestFixture]
    public class TextReporterTests
    {
        [Test]
        [MethodImpl(MethodImplOptions.NoInlining)]
        public void ShouldProduceExpectedReport()
        {
            var stories = new ReportTestData().CreateMixContainingEachTypeOfOutcomeWithOneScenarioPerStory();
            var actual = new StringBuilder();

            foreach (var story in stories)
            {
                var textReporter = new TextReporter();
                textReporter.Process(story);
                actual.AppendLine(textReporter.ToString());
            }

            Approvals.Verify(actual.ToString(), StackTraceScrubber.Scrub);
        }

        [Test]
        [MethodImpl(MethodImplOptions.NoInlining)]
        public void ShouldProduceExpectedTextWithExamples()
        {
            var stories = new ReportTestData().CreateTwoStoriesEachWithOneFailingScenarioAndOnePassingScenarioWithThreeStepsOfFiveMillisecondsAndEachHasTwoExamples();
            var actual = new StringBuilder();

            foreach (var story in stories)
            {
                var textReporter = new TextReporter();
                textReporter.Process(story);
                actual.AppendLine(textReporter.ToString());
            }

            Approvals.Verify(actual.ToString(), StackTraceScrubber.Scrub);
        }
    }
}
