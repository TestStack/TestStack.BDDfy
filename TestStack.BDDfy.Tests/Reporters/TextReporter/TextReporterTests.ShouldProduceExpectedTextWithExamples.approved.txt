Story: Happiness
	As a person
	I want ice cream
	So that I can be happy

Scenario: Example Scenario
	Given a <sign> account balance        
	When the account holder requests money
	Then money <action> dispensed         

Examples: 
| sign     | action |
| positive | is     |
| negative | is not |


Story: Account holder withdraws cash
	As an account holder
	I want to withdraw cash
	So that I can get money when the bank is closed

Scenario: Example Scenario
	Given a <sign> account balance        
	When the account holder requests money
	Then money <action> dispensed         

Examples: 
| sign     | action | Result | Errors                                                                                                |
| positive | is     | Passed |                                                                                                       |
| negative | is not | Failed | Step: Then money <action> dispensed failed with exception: [Boom With New lines] [Details at 1 below] |

Exceptions:
  1. Boom With New lines
   at TestStack.BDDfy.Tests.Reporters.ReportTestData.GetScenarios(Boolean includeFailingScenario, Boolean includeExamples) in ...\ReportTestData.cs


